aptos-move/block-executor/src/proptest_types/types.rs

```diff
@@ -507,236 +537,15 @@ where
-                        assert_none!(val.extract_raw_bytes());
-                    }
-                    assert_eq!(expected_result.1, None);
-                },
-            })
-    }
-
-    // Used for testing, hence the function asserts the correctness conditions within
-    // itself to be easily traceable in case of an error.
-    pub fn assert_output<K>(&self, results: &Result<Vec<Output<K, V>>, usize>) {
-        match (self, results) {
-            (Self::Aborted(i), Err(Error::UserError(idx))) => {
-                assert_eq!(i, idx);
-            },
-            (Self::SkipRest(skip_at, expected_results), Ok(results)) => {
-                // Check_result asserts internally, so no need to return a bool.
-                results
-                    .iter()
-                    .take(*skip_at)
-                    .zip(expected_results.iter())
-                    .for_each(|(Output(_, _, result, _), expected_results)| {
-                        Self::check_result(expected_results, result)
-                    });
-
-                results
-                    .iter()
-                    .skip(*skip_at)
-                    .for_each(|Output(_, _, result, _)| assert!(result.is_empty()))
-            },
-            (Self::ExceedBlockGasLimit(last_committed, expected_results), Ok(results)) => {
-                // Check_result asserts internally, so no need to return a bool.
-                results
-                    .iter()
-                    .take(*last_committed + 1)
-                    .zip(expected_results.iter())
-                    .for_each(|(Output(_, _, result, _), expected_results)| {
-                        Self::check_result(expected_results, result)
-                    });
-
-                results
-                    .iter()
-                    .skip(*last_committed + 1)
-                    .for_each(|Output(_, _, result, _)| assert!(result.is_empty()))
-            },
-            (Self::DeltaFailure(fail_idx, expected_results), Ok(results)) => {
-                // Check_result asserts internally, so no need to return a bool.
-                results
-                    .iter()
-                    .take(*fail_idx)
-                    .zip(expected_results.iter())
-                    .for_each(|(Output(_, _, result, _), expected_results)| {
-                        Self::check_result(expected_results, result)
-                    });
-            },
-            (Self::Success(expected_results), Ok(results)) => results
-                .iter()
-                .zip(expected_results.iter())
-                .for_each(|(Output(_, _, result, _), expected_result)| {
-                    Self::check_result(expected_result, result);
-                }),
-            _ => panic!("Incomparable execution outcomes"),
-        }
+        // TODO: add variety to generating fee statement based on total gas (self.4).
+        FeeStatement::new(1, self.4 / 2, (self.4 + 1) / 2, 0, 0)
```

Comment:
What is `self.4`? The first value (`total_charge_gas_units`) should be the sum of the next 3 values (execution, io, storage gas).

Reply:
that was the total amount of gas decided randomly between 0 and 5 (MAX_GAS const) for this incarnation by proptest. We split it into the two types of gas that are used within the executor.

Good point about the first value - if that's the case, then I'd suggest that the new method should not take the first parameter at all, why not take the currently last 4 arguments and always derive the first one?

Reply:
Because of precision reasons, we keep the total gas separately in the fee statement. Execution / io / storage gas may get round down / up when output by the gas meter, but total gas is computed before the rounding.

Reply:
For the test, we can just assign the total gas as the sum, as you suggested.